packages/store/addon/-private/system/snapshot.ts

```diff
@@ -294,9 +340,9 @@ export default class Snapshot {
     }
 
     if (id) {
-      this._belongsToIds[keyName] = result;
+      this._belongsToIds[keyName] = result as RecordId;
     } else {
-      this._belongsToRelationships[keyName] = result;
+      this._belongsToRelationships[keyName] = result as Snapshot;
     }
```

Comment:
Can we add a comment explaining what makes these casts safe, so that if the code around it changes later we'll at least have some signal that we need to do to make sure the invariants still hold?